test/integration-tests/IntegrationTests.Helpers/TestHelper.cs

```diff
@@ -82,9 +83,27 @@ public Container StartContainer(int traceAgentPort, int webPort)
 
         PowershellHelper.RunCommand($"docker exec {container.Name} curl -v {healthCheckEndpoint}", Output);
 
-        var (standardOutput, _) = PowershellHelper.RunCommand($"Write-Host (Test-NetConnection -ComputerName 'localhost' -Port {webPort}).TcpTestSucceeded", Output);
+        var healthChecks = 0;
+        var healthChecksUrl = $"http://localhost:{webPort}/health-check";
+        var healthChecksPassed = false;
 
-        standardOutput.Should().Contain("True", $"Test connection to the port number {webPort} failed. On this port container service should be available.");
+        do
+        {
+            var (standardOutput, _) = PowershellHelper.RunCommand($"Write-Host (Invoke-WebRequest -Uri {healthChecksUrl}).StatusCode", Output);
+
+            healthChecksPassed = standardOutput.Trim() == "200";
+
+            if (!healthChecksPassed)
+            {
+                healthChecks++;
+
+                // wait for next health check
+                Thread.Sleep(2000 * healthChecks);
```

Comment:
This value is be less and could cause flaky test results. It better to increase the time window or change our approach.

Reply:
The current hypothesis for the flaky tests is that there is a delay between when the port is available (the current check in place) and when the application can respond to requests. Usually that delay is minuscule, but when the system is under a heavier load, that delay can be relatively long.

Reply:
Just an initial value, might need some fine-tuning.